### PR TITLE
initial api for auth done; initial Welcome scene UI needs to be conne…

### PR DESCRIPTION
**Pull request by hurjun1995**
…cted with api

### CODE REVIEW

File: app/modules/auth/utils/validate.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export function validateEmail(email) {
+    var filter = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+
+    if (filter.test(email)) return true;
+
+    return false;
+}
+
+export function validatePassword(password) {
+    if (password.length > 6) return true;
```
Comment (leesw98) [inline app/modules/auth/utils/validate.js]: could you just `return password.length > 6`? or is there a reason behind you doing this?

Comment (leesw98) [reply app/modules/auth/utils/validate.js]: Can be applied to other parts as well.

File: app/modules/auth/utils/validate.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+export function validate(form) {
+    let error = {};
+    let success = true;
+
+    var keys = Object.keys(form);
+    var length = keys.length;
+
+    keys.slice(0, length).map(field => {
```
Comment (leesw98) [inline app/modules/auth/utils/validate.js]: You can also do keys.slice(0)

**Review by bwdmonkey — APPROVED** [review]
.DS_STORE should be added to .gitignore preferrably.
Few weird formatting but fine for now.
Comments I made are just me being nitpicky. No need to reflect it besides the discussion questions I guess.

LGTM I like where it's going right now.

File: app/modules/auth/components/AuthTextInput/AuthTextInput.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import PropTypes from 'prop-types';
+
+import { View } from 'react-native';
+import { Container, Header, Content, Form, Item, Input, Label } from 'native-base';
+import Icon from "react-native-vector-icons/MaterialCommunityIcons";
+import styles from './styles';
+
+class AuthTextInput extends React.Component {
+    render() {
+        const { showLabel, placeholder, autoFocus, onChangeText, secureTextEntry } = this.props;
```
[the remark this answers is not in the document]
Comment (hurjun1995) [reply app/modules/auth/components/AuthTextInput/AuthTextInput.js]: line 11? It's deconstruction I believe; see the link: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

File: app/modules/auth/redux/actions.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import * as t from './actionTypes';
+import * as api from '../api';
+import { auth } from '../../../config/firebase'
+
+import { AsyncStorage } from 'react-native';
+
+export function register(data, successCB, errorCB) {
+    return (dispatch) => {
+        api.register(data, function (success, data, error) {
+            if (success) successCB(data);
```
[the remark this answers is not in the document]
Comment (hurjun1995) [reply app/modules/auth/redux/actions.js]: I actually just followed this syntax from the tutorial, but wouldn't this way be more clear?

File: app/modules/auth/utils/validate.js (hunk 1 above)
Comment (hurjun1995) [reply app/modules/auth/utils/validate.js]: nope I should change

File: app/modules/auth/utils/validate.js (hunk 2 above)
Comment (hurjun1995) [reply app/modules/auth/utils/validate.js]: yeah should change that too

File: app/modules/auth/components/AuthTextInput/AuthTextInput.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (leesw98) [reply app/modules/auth/components/AuthTextInput/AuthTextInput.js]: Got it.

File: app/modules/auth/redux/actions.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (leesw98) [reply app/modules/auth/redux/actions.js]: I am not sure but I guess it's fine for now. We can refactor if needed

**Pull request merged**